[Cursor Test] Move PcComponent out of `chips/` and into `boards/components`

hudson-ayers opened the pull request:
### Pull Request Overview

This PR was entirely authored by Cursor (model: gpt-5). I intentionally did not make any edits to the generated code -- I wanted to use this as a brief exercise to see how capable cursor was of making a basic refactor. I picked the most recently opened issue. I did not review the changes in depth (yet), but made some suggestions in response to the initial changes. I gave the following 5 prompts to get it here:

1. This file contains a component, but Tock components are all supposed to be defined in the boards/ crate. Can you move the component defined in this file into the boards crate in a similar style to how other components are defined, and update boards/qemu_i486_q35/src/main.rs accordingly so that board still builds?

2. The component should go in the boards/components directory alongside other components. Can you move it there, in a file named x86_pc_component.rs?

3. Please do not call static_init!() within finalize(), instead move that into x86_pc_component_static.

4. Thank you. Can you also move serial_port_component_static!() into this component file?

5. Can you just inline the macro rather than having it be a standalone macro? Also, please be sure to delete it from its original location

### Testing Strategy

This pull request has not been tested.


### TODO or Help Wanted

This pull request still careful review and testing.


### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

bradjc commented:
This initial version is not mergable (uses arch-specific crates in components), but this is, I think, close, and it is a very needed change.

hudson-ayers commented on chips/x86_q35/src/chip.rs:
```diff
@@ -159,60 +185,39 @@ impl<'a, const PR: u16> Chip for Pc<'a, PR> {
+    ) -> Self {
+        // CPU/interrupt controller/VGA baseline init
         unsafe {
             x86::init();
             crate::pic::init();
```
For sam4l etc. we do this in main.rs immediately before the call to create the peripherals; but I think doing it in here is effectively no different

hudson-ayers commented on chips/x86_q35/src/chip.rs:
```diff
@@ -159,60 +185,39 @@ impl<'a, const PR: u16> Chip for Pc<'a, PR> {
-pub struct PcComponent<'a> {
-    pd: &'a mut PD,
-    pt: &'a mut PT,
+/// Default x86 PC peripherals commonly present on Q35 machines.
```
```suggestion
/// Default x86 PC peripherals
```

hudson-ayers commented on chips/x86_q35/src/chip.rs:
```diff
@@ -159,60 +185,39 @@ impl<'a, const PR: u16> Chip for Pc<'a, PR> {
         unsafe {
             x86::init();
             crate::pic::init();
-            // Enable the VGA path by building or running with the feature flag, e.g.:
-            //   `cargo run -- -display none`
-            // A plain `make run` / `cargo run` keeps everything on COM1.
-            //
-            // Initialise VGA and clear BIOS text if VGA is enabled
-            // Clear BIOS banner: the real-mode BIOS leaves its text (and the cursor off-screen) in
-            // 0xB8000.  Wiping the full 80×25 buffer gives us a clean screen and a visible cursor
-            // before the kernel prints its first message.
-            // SAFETY: PAGE_DIR is identity-mapped, aligned, and unique
```
Should bring back these comments

hudson-ayers replied:
Done.

hudson-ayers commented on chips/x86_q35/src/chip.rs:
```diff
@@ -159,60 +185,39 @@ impl<'a, const PR: u16> Chip for Pc<'a, PR> {
-            // 0xB8000.  Wiping the full 80×25 buffer gives us a clean screen and a visible cursor
-            // before the kernel prints its first message.
-            // SAFETY: PAGE_DIR is identity-mapped, aligned, and unique
-            let pd: &mut PD = &mut *core::ptr::from_mut(self.pd);
```
Cursor switched this to just be `self.pd`, which I think is correct because in both the old and new versions we already did the `&mut *ptr::addr_of_mut!()` dance in main.rs

hudson-ayers commented:
I rebased this and applied some more changes based on my own review. There were a lot of conflicts in the original with https://github.com/tock/tock/pull/4596 which made rebasing tricky, so I squashed all the original commits to make rebasing simpler. 

I did the rebase manually, not using Cursor, since it was tricky and I wanted to be sure I kept the spirit of https://github.com/tock/tock/pull/4596 .

alevy commented:
@hudson-ayers perhaps the base for the PR is improper or something (it's not a PR against `master`) and there are _tons_ of changes that are irrelevant (and I believe already in `master`).

bradjc reviewed (approved):
I say we merge this. It fixes a reasonably substantial technical debt as we shouldn't have had static_init in chips in the first place.

alevy commented:
@tock/x86-wg

hudson-ayers commented:
Rebased again, I tested the qemu screen app still works. This has gone a month without comment from the x86 wg, I think we should merge it

bradjc commented:
Agreed. Marked last call.
